documentation/resource-definitions.md

```diff
@@ -0,0 +1,118 @@
+# Resource definitions 
+
+## V1 version
+If V1 version of Resource is needed, please add the following in the `definitions` section of swagger spec.
+
+```
+	"Resource": {
+          "properties": {
+            "id": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource Id"
+            },
+            "name": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource name"
+            },
+            "type": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource type"
+            },
+            "location": {
+              "type": "string",
+              "description": "Resource location"
+            },
+            "tags": {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "description": "Resource tags"
+            }
+          },
+          "required": [
+            "location"
+          ],
+          "x-ms-azure-resource": true
+        }
+```
+
+## V2 version
+If V2 version of Resource is needed, please add the following in the `definitions` section of swagger spec.
```

Comment:
Is there an expectation to annotate a V2 resource with x-ms-azure-resource (or x-ms-azure-resource-v2 or similar)?